File: altoq_frontend/src/app/models/category.ts
export interface Category {
  id: number;
  name: string;
  slug: string;
  description?: string;
  icon?: string;
  parent_id?: number;
  created_at: string;
}

export interface CategoryWithProducts extends Category {
  product_count: number;
}

export interface CategoryTree extends Category {
  subcategories: CategoryTree[];
}

File: altoq_backend/app/schemas/category.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime

class CategoryBase(BaseModel):
    name: str
    slug: str
    description: Optional[str] = None
    icon: Optional[str] = None
    parent_id: Optional[int] = None

class CategoryCreate(CategoryBase):
    pass

class CategoryUpdate(BaseModel):
    name: Optional[str] = None
    slug: Optional[str] = None
    description: Optional[str] = None
    icon: Optional[str] = None
    parent_id: Optional[int] = None

class CategoryResponse(CategoryBase):
    id: int
    created_at: datetime

    class Config:
        from_attributes = True

class Category(CategoryBase):
    id: int
    created_at: datetime

    class Config:
        from_attributes = True

class CategoryWithProducts(Category):
    product_count: int

class CategoryTree(Category):
    subcategories: List['CategoryTree'] = []

    class Config:
        from_attributes = True
